projects/packages/waf/src/class-waf-standalone-bootstrap.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Handles generation and deletion of the bootstrap for the standalone WAF mode.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+namespace Automattic\Jetpack\Waf;
+
+use Exception;
+
+/**
+ * Handles the bootstrap.
+ */
+class WafStandaloneBootstrap {
+
+	/**
+	 * Ensures that constants are initialized if this class is used.
+	 */
+	public function __construct() {
+		$this->guard_against_missing_abspath();
+		$this->initialize_constants();
+	}
+
+	/**
+	 * Ensures that this class is not used unless we are in the right context.
+	 *
+	 * @return void
+	 * @throws Exception If we are outside of WordPress.
+	 */
+	private function guard_against_missing_abspath() {
+
+		if ( ! defined( 'ABSPATH' ) ) {
+			throw new Exception( 'Cannot generate the WAF bootstrap if we are not running in WordPress context.' );
+		}
+	}
+
+	/**
+	 * Initializes the constants required for generating the bootstrap, if they have not been initialized yet.
+	 *
+	 * @return void
+	 */
+	private function initialize_constants() {
+		if ( ! defined( 'JETPACK_WAF_DIR' ) ) {
+			define( 'JETPACK_WAF_DIR', trailingslashit( WP_CONTENT_DIR ) . 'jetpack-waf' );
+		}
+		if ( ! defined( 'JETPACK_WAF_WPCONFIG' ) ) {
```

Comment:
Do we need the WP config location? When running in standalone mode, we shouldn't need any WP-specific info (if we do, the check should run after WP is loaded), and in plugin mode we don't need to load wp-config.php ourselves since everything is in the environment already. The reason I ask is that there are hosts where some WP configuration comes from environment variables rather than wp-config.php, so parsing it ourselves won't tell us the info.

Reply:
Good call, it seems I have dragged it a long prematurely, assuming it is used somewhere. But it is not, so I removed that part again. Thanks!